lib/puppet/util/windows/process.rb

```diff
@@ -60,6 +64,26 @@ def get_current_process
   end
   module_function :get_current_process
 
+  def open_process(desired_access, inherit_handle, process_id, &block)
+    phandle = nil
+    inherit = inherit_handle ? FFI::WIN32_TRUE : FFI::WIN32_FALSE
+    begin
+      phandle = OpenProcess(desired_access, inherit, process_id)
+      if phandle == FFI::Pointer::NULL_HANDLE
+        raise Puppet::Util::Windows::Error.new(
+          "OpenProcess(#{desired_access.to_s(8)}, #{inherit}, #{process_id})")
```

Comment:
Are there any security concerns to sending the process ID of puppet to stdout/stderr here?

I have no clue whether or not PID is basically treated like public information on the system or if we need it to stay private.

Reply:
Nope, answered my own question. Since the process ID here would be the PID of the puppet run I was executing (and not the PID listed in the PIDFile) this wouldn't be sensitive

Reply:
That was also not entirely correct. So here's what I think:

This _will_ in fact read the contents of the lockfile and print the process ID listed in the lockfile to stderr/out if the error occurs.

_however_ I think this is safe (even if the PID should be secure) because if a restricted user was trying to grab this information they wouldn't even be able to read the lockfile in the first place (i.e. the file read of the lockfile would fail before we ever got there).

And if you _could_ read the lockfile it doesn't matter what this prints anyway, since you could just go straight there.